### PR TITLE
Merge jnigen

### PR DESCRIPTION
**Pull request by HosseinYousefi**
_Do NOT squash merge!_

Followed the instructions from https://github.com/dart-lang/tools/pull/100.
Closes https://github.com/dart-lang/jnigen/issues/410.

History change commands:

1. Move everything to `pkgs/`
```
git filter-repo --path-rename :pkgs/
```

2. Remove `AUTHORS` and `LICENSE` files as they already exist.
```
git filter-repo --path pkgs/AUTHORS --path pkgs/LICENSE --invert-paths
```

3. Use top-level readme as package:jnigen's readme and remove the top-level one.
```
git filter-repo --path pkgs/jnigen/README.md --invert-path
git filter-repo --path-rename-match pkgs/README.md:pkgs/jnigen/README.md
```

4. Move `.github` related files to their correct path and remove existing files.
```
git filter-repo --path-rename-match pkgs/.github/:.github/
git filter-repo --path .github/dependabot.yaml --path .github/workflows/publish.yaml --invert-path
```

5. Change the workflow name.
```
git filter-repo --path-rename-match .github/workflows/test-package.yaml:.github/workflows/jnigen.yaml
```

6. Apply some minor changes to paths inside the workflow.
```
git filter-repo --replace-text textreplacement
```

Content of `textreplacement`:
```
working-directory: ./==>working-directory: ./pkgs/
path-to-lcov: ./==>path-to-lcov: ./pkgs/
```

7. Add `[jnigen] ` to the beginning of every commit message and change the issue/PR numbers to the correct link in the `jnigen` repo.
```
git filter-repo --replace-message msgreplacement
```

Content of `msgreplacement`:
```
regex:^==>[jnigen] 
[jnigen] [jnigen] ==>[jnigen] 
regex:#(\d+)==>https://github.com/dart-lang/jnigen/issues/\1
```

---

```
 * [new tag]           jni-v0.6.0 -> jni-v0.6.0
 * [new tag]           jni-v0.7.0 -> jni-v0.7.0
 * [new tag]           jnigen-v0.6.0 -> jnigen-v0.6.0
 * [new tag]           jnigen-v0.7.0 -> jnigen-v0.7.0
```

### CODE REVIEW

**Comment by dcharkes** [comment]
Can we review and land some of the steps on the JNIgen repo before doing the history rewrite? It's a bit hard to review changes before hitting the merge button.

Candidates:

* Use top-level readme as package:jnigen's readme and remove the top-level one.
* Change the workflow name.

**Comment by HosseinYousefi** [comment]
> Can we review and land some of the steps on the JNIgen repo before doing the history rewrite? It's a bit hard to review changes before hitting the merge button.
> 
> Candidates:
> 
> * Use top-level readme as package:jnigen's readme and remove the top-level one.

I basically rewrote the history in a way that the top-level readme has always been in `pkgs/jnigen`. So we can just confirm that it's the case right now.

> * Change the workflow name.

Again, I rewrote the history so that the name of the workflow's file is `jnigen.yaml` instead of what it was before.

There is a separate commit with only the new changes ([Make GitHub workflow ready for the merger](https://github.com/dart-lang/native/pull/545/commits/bf2e8d8c80f445d030b4f7caf9ceb8f5f2e4eb06))

I'll try to get the CI working, and if that happens then I think we're in a good spot to merge it.

**Review by dcharkes — APPROVED** [review]

**Pull request merged**
